src/common/test/componentMocking.tsx

```diff
@@ -231,27 +256,29 @@ export const renderComponentWithMocksAndContexts = async (
     }
   };
 
+  const waitForElementAndValue: TestTools['waitForElementAndValue'] = async props => {
+    const { selector, value } = props;
+    return waitFor(async () => {
+      const elementValue = await getTextOrInputValue(selector);
+      if (value && (!elementValue || !elementValue.includes(value))) {
```

Comment:
What if `value` is false in this condition? It seems to not wait for value, but for the elementValue?

Reply:
Value cannot be false, it must be a string. "Value" is inside the "if" to cover the case when it is an empty string ''.

This is quite unclear and I needed some time to find out why it is there. I changed that part to be a lot clearer:

if (value === '' && elementValue === '') {
        return;
      }

ElementValue can be "undefined" until element is found in the dom. That's why !elementValue wont work.

Reply:
Done.